lib/backend/buffer_test.go

```diff
@@ -134,11 +137,69 @@ func (s *BufferSuite) TestWatcherSimple(c *check.C) {
 	}
 }
 
+// TestWatcherCapacity checks various watcher capacity scenarios
+func (s *BufferSuite) TestWatcherCapacity(c *check.C) {
+	const gracePeriod = time.Second
+
+	ctx := context.TODO()
```

Comment:
nit: context.Background instead of TODO - there is no TODO to address, since there is no other context to push here.

```suggestion
	ctx := context.Background()
```